### PR TITLE
Fixed ISymbol equality comparisons

### PR DESCRIPTION
**Pull request by jnm2**
Fixes the broken tests mentioned in https://github.com/DotNetAnalyzers/StyleCopAnalyzers/pull/2758.

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1100DoNotPrefixCallsWithBaseUnlessLocalImplementationExists.cs [hunk 1]
```diff
@@ -112,7 +112,7 @@ private static void HandleBaseExpression(SyntaxNodeAnalysisContext context)
             }
 
             var speculativeSymbol = context.SemanticModel.GetSpeculativeSymbolInfo(parent.SpanStart, speculativeExpression, SpeculativeBindingOption.BindAsExpression);
-            if (speculativeSymbol.Symbol != targetSymbol.Symbol)
+            if (speculativeSymbol.Symbol.Equals(targetSymbol.Symbol))
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1100DoNotPrefixCallsWithBaseUnlessLocalImplementationExists.cs]: ❗️ This should have been negated

File: StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/StandardTextDiagnosticBase.cs [hunk 2]
```diff
@@ -193,7 +193,7 @@ private static bool SeeTagIsCorrect(SyntaxNodeAnalysisContext context, XElement
             }
 
             INamedTypeSymbol expectedSymbol = semanticModel.GetDeclaredSymbol(constructorDeclarationSyntax.Parent, context.CancellationToken) as INamedTypeSymbol;
-            return actualSymbol.OriginalDefinition == expectedSymbol;
+            return actualSymbol.OriginalDefinition.Equals(expectedSymbol);
```
Comment (jnm2) [inline StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/StandardTextDiagnosticBase.cs]: @sharwell Do you happen to know whether `OriginalDefinition` can return null?

Comment (sharwell) [reply StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/StandardTextDiagnosticBase.cs]: I am not certain

**Comment by jnm2** [comment]
@sharwell Ready for review.

File: StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1130UseLambdaSyntax.cs [hunk 3]
```diff
@@ -117,7 +117,7 @@ private static bool HandleMethodInvocation(SemanticModel semanticModel, Anonymou
                 var invocationExpression = originalInvocationExpression.ReplaceNode(anonymousMethod, lambdaExpression);
                 SymbolInfo newSymbolInfo = semanticModel.GetSpeculativeSymbolInfo(location.SourceSpan.Start, invocationExpression, SpeculativeBindingOption.BindAsExpression);
 
-                if (originalSymbolInfo.Symbol != newSymbolInfo.Symbol)
+                if (!originalSymbolInfo.Symbol.Equals(newSymbolInfo.Symbol))
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1130UseLambdaSyntax.cs]: ❔ Are we sure `originalSymbolInfo.Symbol` cannot be null?

Comment (jnm2) [reply StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1130UseLambdaSyntax.cs]: Yes, or else `GetDelegateParameterList` would have thrown NRE due to the call on line 107. If that's too distant of a guarantee, I have no problem using `Equals`.

**Pull request merged**